pkg/types/powervs/validation/platform.go

```diff
@@ -6,17 +6,56 @@ import (
 	"github.com/openshift/installer/pkg/types/powervs"
 )
 
+var powerVSRegions = func() map[string]string {
+	regions := make(map[string]string)
+
+	for _, region := range powervs.Regions {
+		regions[region.Name] = regions[region.Description]
+	}
+	return regions
+}()
+
+var regionShortNames = func() []string {
+	keys := make([]string, len(powerVSRegions))
+	i := 0
+	for r := range powerVSRegions {
+		keys[i] = r
+		i++
+	}
+	return keys
+}()
+
 // ValidatePlatform checks that the specified platform is valid.
 func ValidatePlatform(p *powervs.Platform, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	if p.Region == "" {
 		allErrs = append(allErrs, field.Required(fldPath.Child("region"), "region must be specified"))
+	} else if _, ok := powerVSRegions[p.Region]; !ok {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("region"), p.Region, regionShortNames))
+	}
+
+	allErrs = append(allErrs, validateVPCConfig(p, fldPath)...)
+
+	if p.DefaultMachinePlatform != nil {
+		allErrs = append(allErrs, ValidateMachinePool(p.DefaultMachinePlatform, fldPath.Child("defaultMachinePlatform"))...)
 	}
+	return allErrs
+}
 
-	/*if p.DefaultMachinePlatform != nil {
-		allErrs = append(allErrs, ValidateMachinePool(p, p.DefaultMachinePlatform, fldPath.Child("defaultMachinePlatform"))...)
-	}*/
+func validateVPCConfig(p *powervs.Platform, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if p.VPC != "" || len(p.Subnets) > 0 || p.PowerVSResourceGroup != "" {
+		if p.VPC == "" {
+			allErrs = append(allErrs, field.Required(path.Child("vpc"), "vpc is required when specifying subnets or ResourceGroup"))
+		}
+		if len(p.Subnets) == 0 {
+			allErrs = append(allErrs, field.Required(path.Child("subnets"), "subnets is required when specifying vpc or ResourceGroup"))
+		}
+		if p.PowerVSResourceGroup == "" {
+			allErrs = append(allErrs, field.Required(path.Child("ResourceGroup"), "ResourceGroup is required when specifying vpc or subnets"))
```

Comment:
it's required for vpc, etc., but I don't think it's important that users know what components its used to create.